proxmox/api/credentials.go

```diff
@@ -16,41 +17,61 @@ const rootUsername = "root@pam"
 // Credentials is a struct that holds the credentials for the Proxmox Virtual
 // Environment API.
 type Credentials struct {
-	Username string
-	Password string
-	OTP      *string
-	APIToken *string
+	AuthPayload         string
+	AuthTicket          string
+	CSRFPreventionToken string
+	APIToken            *string
+	OTP                 *string
+	Username            string
+	Password            string
 }
 
 // NewCredentials creates a new Credentials struct.
-func NewCredentials(username, password, otp, apiToken string) (*Credentials, error) {
-	if apiToken != "" {
+func NewCredentials(username, password, otp, apiToken string, authTicket string, csrfPreventionToken string, authPayload string) (*Credentials, error) {
+
+	var errorAuthTicketCommonMsg = "the Proxmox Virtual Environment API requires auth params; "
+
+	switch {
```

Comment:
@bpg im curious if this logic-block would be better within `proxmox/api/client.go` (around L131) ?